src/main/java/org/jabref/logic/pdf/search/indexing/DocumentReader.java

```diff
@@ -0,0 +1,138 @@
+package org.jabref.logic.pdf.search.indexing;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.strings.StringUtil;
+import org.jabref.preferences.FilePreferences;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.document.TextField;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+
+import static org.jabref.model.pdf.search.SearchFieldConstants.CONTENT;
+import static org.jabref.model.pdf.search.SearchFieldConstants.MODIFIED;
+import static org.jabref.model.pdf.search.SearchFieldConstants.PATH;
+
+/**
+ * Utility class for reading the data from LinkedFiles of a BibEntry for Lucene.
+ */
+public final class DocumentReader {
+    private static final Log LOGGER = LogFactory.getLog(DocumentReader.class);
+
+    private final BibEntry entry;
+    private final FilePreferences filePreferences;
+
+    /**
+     * Creates a new DocumentReader using a BibEntry.
+     *
+     * @param bibEntry Must not be null and must have at least one LinkedFile.
+     */
+    public DocumentReader(BibEntry bibEntry, FilePreferences filePreferences) {
+        this.filePreferences = filePreferences;
+        if (bibEntry.getFiles().isEmpty()) {
+            throw new IllegalStateException("There are no linked PDF files to this BibEntry!");
+        }
+
+        this.entry = bibEntry;
+    }
+
+    /**
+     * Reads a LinkedFile of a BibEntry and converts it into a Lucene Document which is then returned.
+     *
+     * @return An Optional of a Lucene Document with the (meta)data. Can be empty if there is a problem reading the LinkedFile.
+     */
+    public Optional<Document> readLinkedPdf(BibDatabaseContext databaseContext, LinkedFile pdf) {
+        Optional<Path> pdfPath = pdf.findIn(databaseContext, filePreferences);
+        if (pdfPath.isPresent()) {
+            try {
+                return Optional.of(readPdfContents(pdf, pdfPath.get()));
+            } catch (IOException e) {
+                LOGGER.info("Could not read pdf file: " + pdf.getLink() + "!", e);
```

Comment:
Shouldn't this be LOGGER.error?

Reply:
See http://www.slf4j.org/faq.html#paramException,the last parameter must be exception then
                LOGGER.info("Could not read pdf file {}!,pdf.getLInk() e);